nixos/modules/services/web-apps/galene.nix

```diff
@@ -0,0 +1,187 @@
+          Galene's data directory where it needs to reside.
+        '';
+      };
+
+      keyFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/path/to/your/key.pem";
+        description = ''
+          Path to the server's private key. The file is copied at runtime to
+          Galene's data directory where it needs to reside.
+        '';
+      };
+
+      httpAddress = mkOption {
+        type = types.str;
+        default = "";
+        description = "HTTP listen address for galene.";
+      };
+
+      httpPort = mkOption {
+        type = types.int;
+        default = 8443;
+        description = "HTTP listen port.";
+      };
+
+      staticRootPath = mkOption {
+        type = types.str;
+        default = "${cfg.package.static}/static";
+        example = "/var/lib/galene/static";
+        description = "Web server directory.";
+      };
+
+      recordingsRootPath = mkOption {
+        type = types.str;
+        default = defaultRecordingsRootPath;
+        example = "/var/lib/galene/recordings";
+        description = "Recordings directory.";
+      };
+
+      dataRootPath = mkOption {
+        type = types.str;
+        default = defaultDataRootPath;
+        example = "/var/lib/galene/data";
+        description = "Data directory.";
+      };
+
+      groupsRootPath = mkOption {
+        type = types.str;
+        default = defaultGroupsRootPath;
+        example = "/var/lib/galene/groups";
+        description = "Web server directory.";
+      };
+
+      package = mkOption {
+        default = pkgs.galene;
+        defaultText = "pkgs.galene";
+        type = types.package;
+        description = ''
+          Package for running Galene.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.insecure || (cfg.certFile != null && cfg.keyFile != null);
+        message = ''
+          Galene needs both certFile and keyFile defined for encryption, or
+          the insecure flag.
+        '';
+      }
+    ];
+
+    systemd.tmpfiles.rules = [ ]
+      ++ optional (!cfg.insecure && cfg.certFile != null && cfg.keyFile != null)
+      ''
+        C '${cfg.dataRootPath}/key.pem' 0700 ${cfg.user} ${cfg.group} - ${cfg.keyFile}
+        C '${cfg.dataRootPath}/cert.pem' 0700 ${cfg.user} ${cfg.group} - ${cfg.certFile}
+      ''
+      ++ optional (cfg.recordingsRootPath == defaultRecordingsRootPath)
+      "d '${cfg.recordingsRootPath}' 0770 ${cfg.user} ${cfg.group}"
+      ++ optional (cfg.dataRootPath == defaultDataRootPath)
+      "d '${cfg.dataRootPath}' 0770 ${cfg.user} ${cfg.group}"
+      ++ optional (cfg.groupsRootPath == defaultGroupsRootPath)
+      "d '${cfg.groupsRootPath}' 0770 ${cfg.user} ${cfg.group}";
+
+    systemd.services.galene = {
+      description = "galene";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.galene ];
```

Comment:
Is this needed? Why?

Reply:
No, I think this is not needed, but that was proposed by @RaitoBezarius , as it would allow to easily change the package at some point if needed, I guess.

Reply:
@RaitoBezarius can you speak to that? I'm not sure how that would help change the package or why it would be needed.

Reply:
I don't think I suggested this @rgrunbla :p
I suggested to change `pkgs.galene` occurrences by `cfg.package` and adding it a module entry so that everyone can change the package actually used, which is very useful when you want to pass a private package or overriden one without using overlays, etc. And afaik, it's quite classic in NixOS modules.
As for the `path` pattern, I have no strong feeling on this, I do not think it is necessary, we just need to do `${cfg.package}/bin/galene` in the `ExecStart` or something akin to this.

Reply:
Oh, yeah, I misread @aanderse comment, sorry. Fixed.